### PR TITLE
Add LruHotCount and index cardinality tracking (SYN-6497)

### PR DESCRIPTION
**Pull request by Cisphyx**

### CODE REVIEW

File: synapse/lib/lmdbslab.py [hunk 1]
```diff
@@ -310,6 +310,77 @@ def set(self, name: str, valu):
     def get(self, name: str, defv=0):
         return self.cache.get(name.encode(), defv)
 
+class LruHotCount(s_base.Base):
+    '''
+    HotCount with size limit and LRU cache for large key sets.
+    '''
+    EncFunc = staticmethod(s_common.signedint64en)
```
Comment (invisig0th) [inline synapse/lib/lmdbslab.py:317]: snake case plz

**Review by invisig0th — CHANGES_REQUESTED** [review]
one aesthetic nit. looks great!

File: synapse/lib/lmdbslab.py (hunk 1 above)
Comment (invisig0th) [reply synapse/lib/lmdbslab.py:317]: Could probably rename `encode` and `decode` as well :)

Comment (Cisphyx) [reply synapse/lib/lmdbslab.py]: updated to encode/decode

**Review by invisig0th — APPROVED** [review]

**Pull request merged**